doc/man7/EVP_KDF-TLS13_KDF.pod

```diff
@@ -0,0 +1,128 @@
+=pod
+
+=head1 NAME
+
+EVP_KDF-TLS13_KDF - The TLS 1.3 EVP_KDF implementation
+
+=head1 DESCRIPTION
+
+Support for computing the TLS 1.3 flavour of the B<HKDF> KDF through
+the B<EVP_KDF> API.
+
+The EVP_KDF-TLS13_KDF algorithm implements the HKDF key derivation function
+as used by TLS 1.3.
+
+=head2 Identity
+
+"TLS13-KDF" is the name for this implementation; it
+can be used with the EVP_KDF_fetch() function.
+
+=head2 Supported parameters
+
+The supported parameters are:
+
+=over 4
+
+=item "properties" (B<OSSL_KDF_PARAM_PROPERTIES>) <UTF8 string>
+
+=item "digest" (B<OSSL_KDF_PARAM_DIGEST>) <UTF8 string>
+
+=item "key" (B<OSSL_KDF_PARAM_KEY>) <octet string>
+
+=item "salt" (B<OSSL_KDF_PARAM_SALT>) <octet string>
+
+These parameters work as described in L<EVP_KDF(3)/PARAMETERS>.
+
+=item "prefix" (B<OSSL_KDF_PARAM_PREFIX>) <octet string>
+
+This parameter sets the label prefix on the specified TLS 1.3 KDF context.
+For TLS 1.3 this should be set to the ASCII string "tls13 " without a
+trailing zero byte.  Refer to RFC 8446 section 7.1 "Key Schedule" for details.
+
+=item "label" (B<OSSL_KDF_PARAM_LABEL>) <octet string>
+
+This parameter sets the label on the specified TLS 1.3 KDF context.
+Refer to RFC 8446 section 7.1 "Key Schedule" for details.
+
+=item "data" (B<OSSL_KDF_PARAM_DATA>) <octet string>
+
+This parameter sets the context data on the specified TLS 1.3 KDF context.
+Refer to RFC 8446 section 7.1 "Key Schedule" for details.
+
+=item "mode" (B<OSSL_KDF_PARAM_MODE>) <UTF8 string> or <integer>
+
+This parameter sets the mode for the TLS 1.3 KDF operation.
+There are two modes that are currently defined:
+
+=over 4
+
+=item B<EVP_KDF_HKDF_MODE_EXTRACT_ONLY> "EXTRACT_ONLY"
```

Comment:
It is quite confusing that the EVP_KDF_HKDF_MODE_EXTRACT_ONLY is the integer constant and not the "EXTRACT_ONLY" string. I see this was copied from the HKDF documentation, maybe it should be somehow fixed there as well?

Reply:
Not sure what to do about these two...

Reply:
What about: `=item "EXTRACT_ONLY" or B<EVP_KDF_HKDF_MODE_EXTRACT_ONLY>` This makes the order of the values the same as of the possible types of the parameter above. And adding the `or` somewhat indicates that these two are actually a different thing.

Reply:
That should work.

I was tempted to not support both types for this param in the TLS 1.3 KDF but sharing the param handling function sort of prevented this.